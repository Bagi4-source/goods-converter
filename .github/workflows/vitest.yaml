name: 'Test'
on:
  pull_request:
  push:

jobs:
  vitest:
    name: Run vitest
    runs-on: ubuntu-latest

    permissions:
      # Required to checkout the code
      contents: read
      # Required to put a comment into the pull-request
      pull-requests: write

    steps:
      - uses: actions/checkout@v2
      - name: 'Install Node'
        uses: actions/setup-node@v2
        with:
          node-version: '18.x'
      - name: 'Install Deps'
        run: npm i
      - name: 'Test'
        run: npm run test
      - name: 'Coverage'
        run: npm run coverage
      - name: 'Report Coverage'
        if: always() # Also generate the report if tests are failing
        uses:  davelosert/vitest-coverage-report-action@v2