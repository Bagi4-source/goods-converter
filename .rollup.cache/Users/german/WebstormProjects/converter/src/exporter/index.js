export * from "./exporter";
export * from "./exporter.types";
//# sourceMappingURL=index.js.map