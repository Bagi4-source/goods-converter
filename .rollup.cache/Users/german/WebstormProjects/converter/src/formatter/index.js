export * from "./CSV.formatter";
export * from "./YML.formatter";
export * from "./formater.types";
//# sourceMappingURL=index.js.map