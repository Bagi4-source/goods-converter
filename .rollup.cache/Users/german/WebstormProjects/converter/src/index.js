export * from "./exporter";
export * from "./formatter";
//# sourceMappingURL=index.js.map