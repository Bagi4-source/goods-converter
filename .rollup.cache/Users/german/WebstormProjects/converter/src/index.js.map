{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,YAAY,CAAA;AAC1B,cAAc,aAAa,CAAA","sourcesContent":["export * from \"./exporter\"\nexport * from \"./formatter\""]}