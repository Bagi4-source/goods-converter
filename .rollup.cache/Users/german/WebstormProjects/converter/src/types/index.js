export * from "./Product.types";
export * from "./Category.types";
//# sourceMappingURL=index.js.map