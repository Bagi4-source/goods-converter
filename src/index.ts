export * from "./exporter"
export * from "./formatter"
export * from "./types"