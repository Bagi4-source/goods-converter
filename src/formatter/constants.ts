import {CSVFormatter} from "./CSV.formatter";
import {YMLFormatter} from "./YML.formatter";
import {TildaFormatter} from "./Tilda.formatter";

export const Formatters = {
    CSV: new CSVFormatter(),
    YML: new YMLFormatter(),
    Tilda: new TildaFormatter(),
}
