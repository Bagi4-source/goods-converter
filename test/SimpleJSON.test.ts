import { expect, describe, it } from "vitest";

import { Formatters } from "../src";
import { brands, categories, products } from "./constants";

describe("Simple JSON formatter", () => {
  const formatter = new Formatters.SimpleJSONFormatter();

  it("should export Simple JSON data", async () => {
    const result = await formatter.format(products, categories, brands);
    expect(result).toMatchSnapshot();
  });
});
