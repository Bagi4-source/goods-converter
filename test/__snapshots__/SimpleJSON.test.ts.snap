// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple JSON formatter > should export Simple JSON data 1`] = `
JsonStreamStringify {
  "_events": {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": StringDecoder {
      "encoding": "utf8",
      Symbol(kNativeDecoder): {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          1,
        ],
        "type": "Buffer",
      },
    },
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": "utf8",
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": false,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "buffer": "",
  "bufferLength": 0,
  "bufferSize": 512,
  "cycle": false,
  "include": undefined,
  "indent": undefined,
  "item": {
    "depth": 1,
    "path": undefined,
    "read": [Function],
    "value": {
      "brands": [
        {
          "id": 1,
          "name": "Adidas",
        },
        {
          "coverURL": "url",
          "id": 2,
          "name": "Nike",
        },
        {
          "id": 3,
          "name": "New Balance",
        },
      ],
      "categories": [
        {
          "id": 1,
          "name": "Обувь",
          "parentId": 2,
        },
        {
          "id": 2,
          "name": "Одежда, обувь и аксессуары",
          "parentId": 3,
        },
        {
          "id": 3,
          "name": "Все товары",
        },
      ],
      "products": [
        {
          "additionalExpenses": 1000,
          "adult": false,
          "age": {
            "unit": "year",
            "value": 6,
          },
          "available": true,
          "barcode": "567890567893",
          "boxCount": 1,
          "categoryId": 1,
          "certificate": "111",
          "children": [
            {
              "additionalExpenses": 1000,
              "adult": false,
              "age": {
                "unit": "year",
                "value": 6,
              },
              "available": true,
              "barcode": "567890567893",
              "boxCount": 1,
              "categoryId": 1,
              "certificate": "111",
              "codesTN": [
                "code",
              ],
              "cofinancePrice": 0,
              "count": 24,
              "countryOfOrigin": "Китай",
              "currency": "RUR",
              "description": "Description",
              "dimensions": "12/32/43",
              "disabled": false,
              "downloadable": false,
              "images": [
                "image1",
                "image2",
              ],
              "manufacturerWarranty": true,
              "minQuantity": 1,
              "oldPrice": 20000,
              "params": [
                {
                  "key": "size",
                  "value": "44",
                },
                {
                  "key": "color",
                  "value": "white",
                },
              ],
              "parentId": 1111,
              "price": 19000,
              "productId": 1,
              "properties": [
                {
                  "key": "seasons",
                  "value": "spring, summer",
                },
              ],
              "purchasePrice": 15000,
              "saleDate": "01.01.2020",
              "seriesName": "Jordan 4",
              "serviceLifeComment": "",
              "serviceLifePeriod": "",
              "stepQuantity": 1,
              "tags": [
                "Nike",
                "Кроссовки",
              ],
              "title": "Title",
              "url": undefined,
              "validityComment": "",
              "validityPeriod": "",
              "variantId": 1112,
              "vat": "VAT_20",
              "vendor": "Nike",
              "vendorCode": "NIKE-1111",
              "vendorId": 2,
              "videos": [],
              "warrantyComment": "",
              "warrantyPeriod": "",
              "weight": 0.15,
            },
          ],
          "codesTN": [
            "code",
          ],
          "cofinancePrice": 0,
          "count": 24,
          "countryOfOrigin": "Китай",
          "currency": "RUR",
          "description": "Description",
          "dimensions": "12/32/43",
          "disabled": false,
          "downloadable": false,
          "gender": "Мужское",
          "images": [
            "image1",
            "image2",
          ],
          "keywords": [
            "Обувь",
            "Кроссовки",
          ],
          "manufacturerWarranty": true,
          "minQuantity": 1,
          "oldPrice": 20000,
          "params": [
            {
              "key": "param1",
              "value": "v1",
            },
            {
              "key": "param2",
              "value": "v2",
            },
          ],
          "price": 19000,
          "productId": 1,
          "properties": [
            {
              "key": "prop1",
              "value": "v1",
            },
            {
              "key": "prop2",
              "value": "v2",
            },
          ],
          "purchasePrice": 15000,
          "relatedProducts": [
            1234,
          ],
          "saleDate": "01.01.2020",
          "seriesName": "Jordan 4",
          "serviceLifeComment": "",
          "serviceLifePeriod": "",
          "sizes": [
            {
              "delimiter": ",",
              "name": "UK",
              "value": "36,37,38,39",
            },
          ],
          "stepQuantity": 1,
          "tags": [
            "Nike",
            "Кроссовки",
          ],
          "title": "Title",
          "url": "http://localhost:8080/?param1=value1&param2=value2",
          "validityComment": "",
          "validityPeriod": "",
          "variantId": 1111,
          "vat": "VAT_20",
          "vendor": "Nike",
          "vendorCode": "NIKE-1111",
          "vendorId": 2,
          "videos": [],
          "warrantyComment": "",
          "warrantyPeriod": "",
          "weight": 0.15,
        },
      ],
    },
    "write": [Function],
  },
  "objectItem": undefined,
  "prePush": undefined,
  "pushCalled": false,
  "readMore": false,
  "readSize": 0,
  "readState": 0,
  "reading": false,
  "replacer": undefined,
  "root": {
    "depth": 0,
    "indent": "",
    "path": [],
    "value": {
      "": {
        "brands": [
          {
            "id": 1,
            "name": "Adidas",
          },
          {
            "coverURL": "url",
            "id": 2,
            "name": "Nike",
          },
          {
            "id": 3,
            "name": "New Balance",
          },
        ],
        "categories": [
          {
            "id": 1,
            "name": "Обувь",
            "parentId": 2,
          },
          {
            "id": 2,
            "name": "Одежда, обувь и аксессуары",
            "parentId": 3,
          },
          {
            "id": 3,
            "name": "Все товары",
          },
        ],
        "products": [
          {
            "additionalExpenses": 1000,
            "adult": false,
            "age": {
              "unit": "year",
              "value": 6,
            },
            "available": true,
            "barcode": "567890567893",
            "boxCount": 1,
            "categoryId": 1,
            "certificate": "111",
            "children": [
              {
                "additionalExpenses": 1000,
                "adult": false,
                "age": {
                  "unit": "year",
                  "value": 6,
                },
                "available": true,
                "barcode": "567890567893",
                "boxCount": 1,
                "categoryId": 1,
                "certificate": "111",
                "codesTN": [
                  "code",
                ],
                "cofinancePrice": 0,
                "count": 24,
                "countryOfOrigin": "Китай",
                "currency": "RUR",
                "description": "Description",
                "dimensions": "12/32/43",
                "disabled": false,
                "downloadable": false,
                "images": [
                  "image1",
                  "image2",
                ],
                "manufacturerWarranty": true,
                "minQuantity": 1,
                "oldPrice": 20000,
                "params": [
                  {
                    "key": "size",
                    "value": "44",
                  },
                  {
                    "key": "color",
                    "value": "white",
                  },
                ],
                "parentId": 1111,
                "price": 19000,
                "productId": 1,
                "properties": [
                  {
                    "key": "seasons",
                    "value": "spring, summer",
                  },
                ],
                "purchasePrice": 15000,
                "saleDate": "01.01.2020",
                "seriesName": "Jordan 4",
                "serviceLifeComment": "",
                "serviceLifePeriod": "",
                "stepQuantity": 1,
                "tags": [
                  "Nike",
                  "Кроссовки",
                ],
                "title": "Title",
                "url": undefined,
                "validityComment": "",
                "validityPeriod": "",
                "variantId": 1112,
                "vat": "VAT_20",
                "vendor": "Nike",
                "vendorCode": "NIKE-1111",
                "vendorId": 2,
                "videos": [],
                "warrantyComment": "",
                "warrantyPeriod": "",
                "weight": 0.15,
              },
            ],
            "codesTN": [
              "code",
            ],
            "cofinancePrice": 0,
            "count": 24,
            "countryOfOrigin": "Китай",
            "currency": "RUR",
            "description": "Description",
            "dimensions": "12/32/43",
            "disabled": false,
            "downloadable": false,
            "gender": "Мужское",
            "images": [
              "image1",
              "image2",
            ],
            "keywords": [
              "Обувь",
              "Кроссовки",
            ],
            "manufacturerWarranty": true,
            "minQuantity": 1,
            "oldPrice": 20000,
            "params": [
              {
                "key": "param1",
                "value": "v1",
              },
              {
                "key": "param2",
                "value": "v2",
              },
            ],
            "price": 19000,
            "productId": 1,
            "properties": [
              {
                "key": "prop1",
                "value": "v1",
              },
              {
                "key": "prop2",
                "value": "v2",
              },
            ],
            "purchasePrice": 15000,
            "relatedProducts": [
              1234,
            ],
            "saleDate": "01.01.2020",
            "seriesName": "Jordan 4",
            "serviceLifeComment": "",
            "serviceLifePeriod": "",
            "sizes": [
              {
                "delimiter": ",",
                "name": "UK",
                "value": "36,37,38,39",
              },
            ],
            "stepQuantity": 1,
            "tags": [
              "Nike",
              "Кроссовки",
            ],
            "title": "Title",
            "url": "http://localhost:8080/?param1=value1&param2=value2",
            "validityComment": "",
            "validityPeriod": "",
            "variantId": 1111,
            "vat": "VAT_20",
            "vendor": "Nike",
            "vendorCode": "NIKE-1111",
            "vendorId": 2,
            "videos": [],
            "warrantyComment": "",
            "warrantyPeriod": "",
            "weight": 0.15,
          },
        ],
      },
    },
  },
  "visited": [
    {
      "brands": [
        {
          "id": 1,
          "name": "Adidas",
        },
        {
          "coverURL": "url",
          "id": 2,
          "name": "Nike",
        },
        {
          "id": 3,
          "name": "New Balance",
        },
      ],
      "categories": [
        {
          "id": 1,
          "name": "Обувь",
          "parentId": 2,
        },
        {
          "id": 2,
          "name": "Одежда, обувь и аксессуары",
          "parentId": 3,
        },
        {
          "id": 3,
          "name": "Все товары",
        },
      ],
      "products": [
        {
          "additionalExpenses": 1000,
          "adult": false,
          "age": {
            "unit": "year",
            "value": 6,
          },
          "available": true,
          "barcode": "567890567893",
          "boxCount": 1,
          "categoryId": 1,
          "certificate": "111",
          "children": [
            {
              "additionalExpenses": 1000,
              "adult": false,
              "age": {
                "unit": "year",
                "value": 6,
              },
              "available": true,
              "barcode": "567890567893",
              "boxCount": 1,
              "categoryId": 1,
              "certificate": "111",
              "codesTN": [
                "code",
              ],
              "cofinancePrice": 0,
              "count": 24,
              "countryOfOrigin": "Китай",
              "currency": "RUR",
              "description": "Description",
              "dimensions": "12/32/43",
              "disabled": false,
              "downloadable": false,
              "images": [
                "image1",
                "image2",
              ],
              "manufacturerWarranty": true,
              "minQuantity": 1,
              "oldPrice": 20000,
              "params": [
                {
                  "key": "size",
                  "value": "44",
                },
                {
                  "key": "color",
                  "value": "white",
                },
              ],
              "parentId": 1111,
              "price": 19000,
              "productId": 1,
              "properties": [
                {
                  "key": "seasons",
                  "value": "spring, summer",
                },
              ],
              "purchasePrice": 15000,
              "saleDate": "01.01.2020",
              "seriesName": "Jordan 4",
              "serviceLifeComment": "",
              "serviceLifePeriod": "",
              "stepQuantity": 1,
              "tags": [
                "Nike",
                "Кроссовки",
              ],
              "title": "Title",
              "url": undefined,
              "validityComment": "",
              "validityPeriod": "",
              "variantId": 1112,
              "vat": "VAT_20",
              "vendor": "Nike",
              "vendorCode": "NIKE-1111",
              "vendorId": 2,
              "videos": [],
              "warrantyComment": "",
              "warrantyPeriod": "",
              "weight": 0.15,
            },
          ],
          "codesTN": [
            "code",
          ],
          "cofinancePrice": 0,
          "count": 24,
          "countryOfOrigin": "Китай",
          "currency": "RUR",
          "description": "Description",
          "dimensions": "12/32/43",
          "disabled": false,
          "downloadable": false,
          "gender": "Мужское",
          "images": [
            "image1",
            "image2",
          ],
          "keywords": [
            "Обувь",
            "Кроссовки",
          ],
          "manufacturerWarranty": true,
          "minQuantity": 1,
          "oldPrice": 20000,
          "params": [
            {
              "key": "param1",
              "value": "v1",
            },
            {
              "key": "param2",
              "value": "v2",
            },
          ],
          "price": 19000,
          "productId": 1,
          "properties": [
            {
              "key": "prop1",
              "value": "v1",
            },
            {
              "key": "prop2",
              "value": "v2",
            },
          ],
          "purchasePrice": 15000,
          "relatedProducts": [
            1234,
          ],
          "saleDate": "01.01.2020",
          "seriesName": "Jordan 4",
          "serviceLifeComment": "",
          "serviceLifePeriod": "",
          "sizes": [
            {
              "delimiter": ",",
              "name": "UK",
              "value": "36,37,38,39",
            },
          ],
          "stepQuantity": 1,
          "tags": [
            "Nike",
            "Кроссовки",
          ],
          "title": "Title",
          "url": "http://localhost:8080/?param1=value1&param2=value2",
          "validityComment": "",
          "validityPeriod": "",
          "variantId": 1111,
          "vat": "VAT_20",
          "vendor": "Nike",
          "vendorCode": "NIKE-1111",
          "vendorId": 2,
          "videos": [],
          "warrantyComment": "",
          "warrantyPeriod": "",
          "weight": 0.15,
        },
      ],
    },
  ],
  Symbol(kCapture): false,
}
`;
